src/lld/backend_patterns/repository/template/Correction.md

```diff
@@ -4,6 +4,76 @@
 # DTO question
 - Q: What is the best practises for DTO objects for 2 or more entity relation?
 - A: DTO can have builder pattern to build required inputs if it seems to more combinations
+- Q: Is Deserialization of json to builder possible?
+- A: Yes, no arg constructor and static builder class is a way for parser.
+- Q: Example
+A: 
+```
+>> invoking code on deserialization
+ObjectMapper mapper = new ObjectMapper();
+Person person = mapper.readValue(jsonString, Person.Builder.class).build();
```

Comment:
ObjectMapper is good. How it is to added to class? ObjectMapper is usually heavy dependency: 

From ChatGPT "If you're using a framework that supports DI, such as Spring or Guice, then you may choose to add the ObjectMapper as a DI component. This can make it easier to manage the lifecycle of the ObjectMapper instance and to inject it into other classes that need to use it."

Reply:
We can have separate template session using modern framework(Spring) to get more visibility around intial setup.

Reply:
okay, in that case, just add a note that frameworks use ObjectMapper is Di as it is computational and resource intensive